### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by MisterDA**
I had to add `result` because of (I think) a conflict with `Container.Result`. If someone manages to fix that, that would be nice.

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
I don't understand what does this fixes. Could you give the error message that you got?

**Comment by kit-ty-kate** [comment]
Also could you remove the opam file generation part? I prefer not to use that.

**Comment by MisterDA** [comment]
Yes, I switched `implicit_transitive_deps` to false and got

```
File "lib/server_configfile.ml", line 213, characters 28-64:
213 |   let yaml = Result.get_exn (Yaml.to_string (yaml_of_conf conf)) in
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This expression has type
         string Yaml.res = (string, Rresult.R.msg) Result.result
       but an expression was expected of type ('a, 'b) result
File "server/backend/backend.ml", line 86, characters 10-29:
86 |     ~mode:(`TCP (`Port port))
               ^^^^^^^^^^^^^^^^^^^
Error: This expression has type [> `TCP of [> `Port of 'a ] ]
       but an expression was expected of type Conduit_lwt_unix.server
       Conduit_lwt_unix.server is abstract because no corresponding cmi file was found in path.
File "server/backend/check.ml", line 33, characters 4-9:
33 |   | Ok () ->
         ^^^^^
Error: This pattern matches values of type ('a, 'b) result
       but a pattern was expected which matches values of type
         (unit, Capnp_rpc.Exception.t) Result.result
       Result.result is abstract because no corresponding cmi file was found in path.
File "server/lib/server.ml", line 130, characters 12-31:
130 |       ~mode:(`TCP (`Port port))
                  ^^^^^^^^^^^^^^^^^^^
Error: This expression has type [> `TCP of [> `Port of 'a ] ]
       but an expression was expected of type Conduit_lwt_unix.server
       Conduit_lwt_unix.server is abstract because no corresponding cmi file was found in path.
 ```
 
 > Also could you remove the opam file generation part? I prefer not to use that.
 
 I figured, it's removed.

**Comment by MisterDA** [comment]
I fixed  the yaml error with the `Yaml.to_string_exn` function, the `` `TCP`` by requiring `conduit-lwt-unix` but the result I couldn't figure it out. The only fix I've found was to require `result` or disable `Containers` for that file, I think.

**Comment by kit-ty-kate** [comment]
is there any chance you could reformat everything like it was before so that i can actually see what has changed? I don't think i can blindly merge otherwise (same about removing the changes of build commands in the opam file)

File: server/main.ml [hunk 1]
```diff
@@ -14,7 +14,7 @@ let info =
     ~man:[`P "This program takes a work directory where every files created \
               are stored. This includes logs, config file and user private \
               keys."]
-    ~version:Config.version
-    Config.name
+    ?version:(Build_info.V1.(Option.map Version.to_string (version ())))
```
Comment (kit-ty-kate) [inline server/main.ml:17]: %{version:pkg} and the version from dune-build-info is not the same value. What's the reason behind this change?

Comment (MisterDA) [reply server/main.ml:17]: My understanding is that [dune-build-info](https://dune.readthedocs.io/en/stable/dune-libs.html#build-info) would provide a nicer version string with the git commit id when opam-health-check installed from git, rather than `0.1~` taken from the opam file. You don't seem interested, so I'll simply remove the commit.

**Comment by MisterDA** [comment]
I've cleaned the PR, I think it's much clearer right now. If you see how to delete the "result" dependency, that would be perfect.

**Comment by kit-ty-kate** [comment]
Superseded by #18 Thanks!
